cli/src/semgrep/config_resolver.py

```diff
@@ -192,15 +196,20 @@ def _load_config_from_local_path(self) -> Dict[str, YamlTree]:
 
     def _make_config_request(self) -> str:
         app_session = get_state().app_session
-        r = app_session.get(
+        resp = app_session.get(
             self._config_path,
             headers={"Accept": "application/json", **self._extra_headers},
         )
-        if r.status_code == requests.codes.ok:
-            return r.content.decode("utf-8", errors="replace")
+        if resp.status_code == requests.codes.ok:
+            try:
+                rule_config = resp.json()["rule_config"]
+                assert isinstance(rule_config, str)
+                return rule_config
+            except BaseException:  # catch JSONDecodeError, AssertionError, etc.
```

Comment:
It's recommended to use Exception instead of BaseException so we don't catch things like keyboard interrupts or other signals

Reply:
```suggestion
            except Exception:  # catch JSONDecodeError, AssertionError, etc.
```